File: pendo_mobile/lib/features/gifts/domain/models/gift.dart
class Gift {
  final int id;
  final String name;
  final String description;
  final String imageUrl;
  final int price;
  final bool isPremium;
  final bool isAnimated;

  Gift({
    required this.id,
    required this.name,
    required this.description,
    required this.imageUrl,
    required this.price,
    required this.isPremium,
    required this.isAnimated,
  });

  factory Gift.fromJson(Map<String, dynamic> json) {
    return Gift(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      imageUrl: json['image_url'],
      price: json['price'],
      isPremium: json['is_premium'],
      isAnimated: json['is_animated'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'image_url': imageUrl,
      'price': price,
      'is_premium': isPremium,
      'is_animated': isAnimated,
    };
  }
}

File: backend/app/schemas/gift.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel


class GiftBase(BaseModel):
    name: str
    description: Optional[str] = None
    image_url: str
    animation_url: Optional[str] = None
    coins: int
    is_animated: bool = False
    is_premium: bool = False
    is_limited: bool = False
    available_until: Optional[datetime] = None
    metadata: Optional[dict] = None


class GiftCreate(GiftBase):
    pass


class Gift(GiftBase):
    id: str
    created_at: datetime

    class Config:
        orm_mode = True


class GiftTransactionBase(BaseModel):
    sender_id: str
    receiver_id: str
    gift_id: str
    coins: int
    message: Optional[str] = None
    is_anonymous: bool = False


class GiftTransactionCreate(GiftTransactionBase):
    pass


class GiftTransaction(GiftTransactionBase):
    id: str
    sent_at: datetime

    class Config:
        orm_mode = True
